tools/LogCollector/source/omiHighCPUDiagnostics.sh

```diff
@@ -0,0 +1,207 @@
+        scx_log_line=$scx_log_line1
+    fi
+
+
+    echo -e "\n\n" >>$omiagent_trace_path
+    echo "SCX log Diff in last 10 min">>$omiagent_trace_path
+    echo "------------------------------------------------">>$omiagent_trace_path
+    omiagent_log_line1=`wc -l $omiagent_log_file |awk '{print $1}'`
+    if [ $omiagent_log_line1 -ne $omiagent_log_line ]; then
+        lineNo=`expr $omiagent_log_line1 - $omiagent_log_line`
+        tail -n $lineNo $omiagent_log_file>>$omiagent_trace_path
+        omiagent_log_line=$omiagent_log_line1
+    fi
+}
+
+
+while [ $count -gt 0 ]
+do
+	date>>$omiagent_trace_path
+    ppuc=(`top -b -n1|grep omiagent|tee -a $omiagent_trace_path|awk '{print $9;print $1}'`)
+
+	index=0
+	index1=0;
+	while [ $index -lt ${#ppuc[@]} ]
+	do
+ 		if [ $(echo "${ppuc[$index]} > $cpu_threshold"|bc) -eq 1 ]; then
+            ind=`expr $index + 1`
+            echo -e "\n\n" >>$omiagent_trace_path
+            echo "Threads with High CPU Utilization:">>$omiagent_trace_path
+            echo "------------------------------------------------">>$omiagent_trace_path
+            pt=(`top -b -H -p ${ppuc[$ind]} -n1|grep omiagent|tee -a $omiagent_trace_path|awk '{print $9;print $1}'`)
+
+            echo -e "\n\n" >>$omiagent_trace_path
+            echo "Stacktrace:">>$omiagent_trace_path
+            echo "----------------------------------------------">>$omiagent_trace_path
+			echo -e "\n\n" >>$omiagent_trace_path
+            echo "Stacktrace for Process: ${ppuc[$ind]}">>$omiagent_trace_path
+            echo "----------------------------------------------">>$omiagent_trace_path
+            sudo gdb -p ${ppuc[$ind]} -batch -ex "thread apply all bt" -ex quit>>$omiagent_trace_path
+
+            while [ $index1 -lt ${#pt[@]} ]
+            do
+                if [ $(echo "${pt[$index1]} > $cpu_threshold"|bc) -eq 1 ]; then
+                    ind1=`expr $index1 + 1`
+                    echo -e "\n\n" >>$omiagent_trace_path
+                    count1=0
+                    echo "Stacktrace for Thread:${pt[$ind1]}">>$omiagent_trace_path
+                    echo "----------------------------------------------">>$omiagent_trace_path
+                    while [ $count1 -lt 5 ]
+                    do
+                        echo -e "\n\n" >>$omiagent_trace_path
+                        echo "Stacktrace Snap Count: $count1">>$omiagent_trace_path
+                        echo "----------------------------------------------">>$omiagent_trace_path
+                        sudo gdb -p ${pt[$ind1]} -batch -ex "bt" -ex quit>>$omiagent_trace_path
+                        count1=`expr $count1 + 1`
+                        sleep 2
+                    done
+                    dump_scx_omi_log_diff_tail
+                fi
+                index1=`expr $index1 + 2`
+            done
+  		fi
+ 		index=`expr $index + 2`
+	done  
+
+	sleep $rescan_interval_sec
+	count=`expr $count - 1`
+done 
+
```

Comment:
add some message informing user where the output file is

Reply:
done